### PR TITLE
Improve calculator

### PR DESCRIPTION
**Pull request by NikitaOstrovsky636**

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -1,26 +1,36 @@
...
-                </div>
-                <div class="calculator-form__action-bar action-bar"> 
-                   <input type="button" class="action-bar__button action-bar__button-sum" id="calc-sam" onclick="calcSum();" value="Sum">
-                   <input type="button" class="action-bar__button action-bar__button-sub" id="calc-sub" onclick="calcSub();" value="Sub">
-                   <input type="button" class="action-bar__button action-bar__button-mul" id="calc-mul" onclick="calcMul();" value="Mul">
-                   <input type="button" class="action-bar__button action-bar__button-div" id="calc-div" onclick="calcDiv();" value="Div">
-                </div>
-            </form>
-            <div class="calculator-result" id="result" ></div>
+            <div class="calculator-form action-bar">
```
Comment (alexeymagdich-tigera) [inline index.html:11]: I'm not sure that `action-bar` is the appropriate class name for this `div`. Please, rethink class names for that form.

File: scss/style.scss [hunk 2]
```diff
@@ -1,23 +1,138 @@
+body {
+    font-family: 'Open Sans', sans-serif;
+    background-color: #f0ecec;
+    padding-top: 50px;
+}
+
+.calculator {
+
```
Comment (alexeymagdich-tigera) [inline scss/style.scss:8]: Please, remove empty line.

File: scss/style.scss [hunk 3]
```diff
@@ -1,23 +1,138 @@
...
+    }
+
+    &__button-sum {
+        grid-area: sum;
+    }
 
+    &__button-result {
+        grid-area: result;
+    }
+}
```
Comment (alexeymagdich-tigera) [inline scss/style.scss:137]: Please, always leave only one empty line at the end of the file.

File: index.html [hunk 4]
```diff
@@ -1,26 +1,36 @@
...
+                <input type="button" class="calculator-form__item operators action-bar__button-sub" id="sub" value="-">
+                <input type="button" class="calculator-form__item number action-bar__button-zero" id="zero" value="0">
+                <input type="button" class="calculator-form__item operators action-bar__button-sum" id="sum" value="+">
+                <input type="button" class="calculator-form__item action-bar__button-result" id="result" value="=">
+            </div>
+            <div class="calculator-bottom">
+                <a class="calculator-bottom__href" href="https://github.com/NikitaOstrovsky636">GitHub</a>
+            </div>
         </div>
+        <script src="./js/script.js"></script>
```
Comment (alexeymagdich-tigera) [inline index.html:34]: Why did you decide to move script loading to the bottom?

File: index.html [hunk 5]
```diff
@@ -1,26 +1,36 @@
...
+                <input type="button" class="calculator-form__item number action-bar__button-one" id="one" value="1">
+                <input type="button" class="calculator-form__item number action-bar__button-two" id="two" value="2">
+                <input type="button" class="calculator-form__item number action-bar__button-three" id="three" value="3">
+                <input type="button" class="calculator-form__item operators action-bar__button-sub" id="sub" value="-">
+                <input type="button" class="calculator-form__item number action-bar__button-zero" id="zero" value="0">
+                <input type="button" class="calculator-form__item operators action-bar__button-sum" id="sum" value="+">
+                <input type="button" class="calculator-form__item action-bar__button-result" id="result" value="=">
+            </div>
+            <div class="calculator-bottom">
+                <a class="calculator-bottom__href" href="https://github.com/NikitaOstrovsky636">GitHub</a>
```
Comment (alexeymagdich-tigera) [inline index.html:31]: Link to GitHub should be opened in a new tab (it was marked in requirements in the Trello ticket).

File: scss/style.scss [hunk 6]
```diff
@@ -1,23 +1,138 @@
+body {
```
Comment (alexeymagdich-tigera) [inline scss/style.scss:1]: I would suggest improving the calculator color palette because currently, it looks little bit scary :grinning: You can use already existing tools for it (e.g. [Coolors](https://coolors.co/)).

File: scss/style.scss [hunk 7]
```diff
@@ -1,23 +1,138 @@
...
+    font-family: 'Open Sans', sans-serif;
+    background-color: #f0ecec;
+    padding-top: 50px;
+}
+
+.calculator {
+    
+    width: 330px;
+    height: 430px;
+    background-color: #797979;
```
Comment (alexeymagdich-tigera) [inline scss/style.scss:11]: I see that you use `#797979` color three times. I would suggest extracting it to separate variable (e.g. `$darkGrey`) and use it where needed. See SASS docs: https://sass-lang.com/documentation/variables.

File: scss/style.scss [hunk 8]
```diff
@@ -1,23 +1,138 @@
...
+    margin: 0 auto;
+    border-radius: 10px 10px 5px 5px;
+    border: 3px solid rgb(30, 4, 77);
```
Comment (alexeymagdich-tigera) [inline scss/style.scss:14]: I see that you use `rgb(30, 4, 77)` color four times. I would suggest extracting it to separate variable (e.g. `$black`) and use it where needed. See SASS docs: https://sass-lang.com/documentation/variables.

File: index.html [hunk 9]
```diff
@@ -1,26 +1,36 @@
...
-                   <input type="button" class="action-bar__button action-bar__button-sub" id="calc-sub" onclick="calcSub();" value="Sub">
-                   <input type="button" class="action-bar__button action-bar__button-mul" id="calc-mul" onclick="calcMul();" value="Mul">
-                   <input type="button" class="action-bar__button action-bar__button-div" id="calc-div" onclick="calcDiv();" value="Div">
-                </div>
-            </form>
-            <div class="calculator-result" id="result" ></div>
+            <div class="calculator-form action-bar">
+                <input type="text" class="calculator-form__item action-bar__output" id="calc-output"  disabled>
+                <input type="button" class="calculator-form__item action-bar__button-ce" id="CE" value="CE">
+                <input type="button" class="calculator-form__item number action-bar__button-seven" id="seven" value="7">
```
Comment (alexeymagdich-tigera) [inline index.html:14]: As I can see, `number` class is not used in `styles.scss` file so you can safely remove it.

File: index.html [hunk 10]
```diff
@@ -1,26 +1,36 @@
...
+                <input type="button" class="calculator-form__item operators action-bar__button-mul" id="mul" value="&times">
+                <input type="button" class="calculator-form__item number action-bar__button-one" id="one" value="1">
+                <input type="button" class="calculator-form__item number action-bar__button-two" id="two" value="2">
+                <input type="button" class="calculator-form__item number action-bar__button-three" id="three" value="3">
+                <input type="button" class="calculator-form__item operators action-bar__button-sub" id="sub" value="-">
+                <input type="button" class="calculator-form__item number action-bar__button-zero" id="zero" value="0">
+                <input type="button" class="calculator-form__item operators action-bar__button-sum" id="sum" value="+">
+                <input type="button" class="calculator-form__item action-bar__button-result" id="result" value="=">
+            </div>
+            <div class="calculator-bottom">
```
Comment (alexeymagdich-tigera) [inline index.html:30]: I think `calculator-footer` is a better name for it.

File: index.html (hunk 5 above)
Comment (alexeymagdich-tigera) [reply index.html:31]: Actually `calculator-bottom__href` is not the best name for it. `href` is an implementation detail. What if we decide to change `a` to `button`? Then `calculator-bottom__href`` will be an incorrect name. 

Please, rename it to `calculator-bottom__github`. That name describes the semantic of the element (link to GitHub), but not the implementation (tag `<a />`).

File: index.html [hunk 11]
```diff
@@ -1,26 +1,36 @@
...
+                <input type="text" class="calculator-form__item action-bar__output" id="calc-output"  disabled>
+                <input type="button" class="calculator-form__item action-bar__button-ce" id="CE" value="CE">
+                <input type="button" class="calculator-form__item number action-bar__button-seven" id="seven" value="7">
+                <input type="button" class="calculator-form__item number action-bar__button-eight" id="eight" value="8">
+                <input type="button" class="calculator-form__item number action-bar__button-nine" id="nine" value="9">
+                <input type="button" class="calculator-form__item operators action-bar__button-div" id="div" value="&divide">
+                <input type="button" class="calculator-form__item number action-bar__button-four" id="four" value="4">
+                <input type="button" class="calculator-form__item number action-bar__button-five" id="five" value="5">
+                <input type="button" class="calculator-form__item number action-bar__button-six" id="six" value="6">
+                <input type="button" class="calculator-form__item operators action-bar__button-mul" id="mul" value="&times">
```
Comment (alexeymagdich-tigera) [inline index.html:21]: As I can see, `operators` class is not used in `styles.scss` file so you can safely remove it.

**Review by alexeymagdich-tigera — CHANGES_REQUESTED** [review]

File: index.html (hunk 1 above)
Comment (alexeymagdich-tigera) [inline index.html]: I'm not sure that `action-bar` is the appropriate class name for this `div`. Please, rethink class names for that form.

File: index.html (hunk 9 above)
Comment (alexeymagdich-tigera) [inline index.html]: As I can see, `number` class is not used in `styles.scss` file so you can safely remove it.

File: index.html (hunk 11 above)
Comment (alexeymagdich-tigera) [inline index.html]: As I can see, `operators` class is not used in `styles.scss` file so you can safely remove it.

File: index.html (hunk 10 above)
Comment (alexeymagdich-tigera) [inline index.html]: I think `calculator-footer` is a better name for it.

File: index.html (hunk 5 above)
Comment (alexeymagdich-tigera) [inline index.html]: Link to GitHub should be opened in a new tab (it was marked in requirements in the Trello ticket).

Comment (alexeymagdich-tigera) [inline index.html]: Actually `calculator-bottom__href` is not the best name for it. `href` is an implementation detail. What if we decide to change `a` to `button`? Or what if we decide to add a few more links (e.g. to Trello etc.)? Then `calculator-bottom__href` will become an incorrect name. 

Please, rename it to `calculator-bottom__github-link`. That name describes the semantic of the element (link to GitHub), but not the implementation (tag `<a />` with `href` attribute).

File: index.html (hunk 4 above)
Comment (alexeymagdich-tigera) [inline index.html:39]: Why did you decide to move script loading to the bottom?

File: scss/style.scss (hunk 6 above)
Comment (alexeymagdich-tigera) [inline scss/style.scss:7]: I would suggest improving the calculator color palette because currently, it looks a little bit scary :grinning: You can use already existing tools for it (e.g. [Coolors](https://coolors.co/)).

File: scss/style.scss (hunk 2 above)
Comment (alexeymagdich-tigera) [inline scss/style.scss]: Please, remove empty line.

File: scss/style.scss (hunk 7 above)
Comment (alexeymagdich-tigera) [inline scss/style.scss]: I see that you use `#797979` color three times. I would suggest extracting it to separate variable (e.g. `$darkGrey`) and use it where needed. See SASS docs: https://sass-lang.com/documentation/variables.

File: scss/style.scss (hunk 8 above)
Comment (alexeymagdich-tigera) [inline scss/style.scss]: I see that you use `rgb(30, 4, 77)` color four times. I would suggest extracting it to separate variable (e.g. `$black`) and use it where needed. See SASS docs: https://sass-lang.com/documentation/variables.

File: scss/style.scss (hunk 3 above)
Comment (alexeymagdich-tigera) [inline scss/style.scss:155]: Please, always leave only one empty line at the end of the file.

File: index.html (hunk 1 above)
Comment (alexeymagdich-tigera) [reply index.html]: You can use the following BEM-structure:

```
-> .calculator-container
----> .calculator
--------> .calculator__output
--------> .calculator__action-bar
------------> .calculator-action-bar__item
----> .calculator-footer
```

File: index.html (hunk 9 above)
Comment (alexeymagdich-tigera) [inline index.html]: I think it would be better to wrap digit buttons to a container (`.calculator__digit-panel`), but it's up to you (depends on how you will re-structure BEM).

Comment (NikitaOstrovsky636) [reply index.html]: I think it would be difficult to use grid if I wrap digit buttons to a container

File: index.html [hunk 12]
```diff
@@ -2,35 +2,36 @@
 <html>
     <head>
         <meta charset="UTF-8">
-        <meta name="viewport" content="width=device-width, initial-scale=1.0">
+        <meta name="viewport" content="width=device-width, initial-scale=action-bar-item__.0">
```
Comment (alexeymagdich) [inline index.html]: Why did you change `initial-scale` value? It's incorrect, see DevTools warning:

image

File: index.html [hunk 13]
```diff
@@ -2,35 +2,36 @@
...
+                    <input type="button" class="calculator-action-bar__item action-bar-item__sub" value="-">
+                    <input type="button" class="calculator-action-bar__item action-bar-item__zero" value="0">
+                    <input type="button" class="calculator-action-bar__item action-bar-item__sum" value="+">
+                    <input type="button" class="calculator-action-bar__item action-bar-item__result" value="=">
+                </div>
             </div>
-            <div class="calculator-bottom">
-                <a class="calculator-bottom__href" href="https://github.com/NikitaOstrovsky636">GitHub</a>
+            <div class="calculator__footer">
+                <a class="calculator-footer__github-link" target="blanc" href="https://github.com/NikitaOstrovsky636">GitHub</a>
```
Comment (alexeymagdich) [inline index.html]: I'm pretty sure `blanc` is an incorrect value for `target` attribute. Did you mean `_blank`?

File: scss/style.scss [hunk 14]
```diff
@@ -1,138 +1,147 @@
...
+        background-color: $alabaster;
+        color: $indigoDye;
         border-radius: 6px;
         font-size: 18pt; 
         font-weight: 700;
         cursor: pointer;
-        box-shadow: 0 0 5px 1px rgb(30, 4, 77);
         border:none;
+
         &:hover {
```
Comment (alexeymagdich) [inline scss/style.scss:68]: Did you test hover? Now it looks odd. When I hover any button its text (`+`, `-`, etc.) is painted over `$indigoDye` color.

File: scss/style.scss [hunk 15]
```diff
@@ -1,138 +1,147 @@
...
-    
+.calculator-container {
     width: 330px;
     height: 430px;
-    background-color: #797979;
+    background-color: $alabaster;
     margin: 0 auto;
-    border-radius: 10px 10px 5px 5px;
-    border: 3px solid rgb(30, 4, 77);
+    padding: 0px 10px;
```
Comment (alexeymagdich) [inline scss/style.scss]: Please, add top padding to calculator and bottom margin to output. I think it will be nicer then.

Untitled

File: scss/style.scss [hunk 16]
```diff
@@ -1,138 +1,147 @@
...
+    border-radius: 0px 0px 10px 10px;
+
+    &__output {
+        box-sizing: border-box;
+        width: 100%;
+        height: 65px;
+        margin: 0px 0px 9px 0px;
+        color: $alabaster;
+        background-color: $indigoDye;
+        font-size: 24pt;
```
Comment (alexeymagdich) [inline scss/style.scss]: Why do you use `pt` here? Choose either `px` or `pt`.

**Review by alexeymagdich — CHANGES_REQUESTED** [review]

File: index.html (hunk 9 above)
Comment (alexeymagdich-tigera) [reply index.html]: @NikitaOstrovsky636 it's not an argument. But for now, I'm OK with that markup.

File: index.html [hunk 17]
```diff
@@ -9,29 +9,30 @@
     <body>
         <div class="calculator-container">
             <div class="calculator">
-                <input type="text" class="calculator__output" disabled>
+                <input type="text" class="calculator__output" id="output" maxlength="6" disabled>
```
Comment (alexeymagdich) [inline index.html]: Why did you add `maxlength` attribute with value of `6`?

File: index.html [hunk 18]
```diff
@@ -9,29 +9,30 @@
...
-                    <input type="button" class="calculator-action-bar__item action-bar-item__mul" value="&times">
-                    <input type="button" class="calculator-action-bar__item action-bar-item__one" value="1">
-                    <input type="button" class="calculator-action-bar__item action-bar-item__two" value="2">
-                    <input type="button" class="calculator-action-bar__item action-bar-item__three" value="3">
-                    <input type="button" class="calculator-action-bar__item action-bar-item__sub" value="-">
-                    <input type="button" class="calculator-action-bar__item action-bar-item__zero" value="0">
-                    <input type="button" class="calculator-action-bar__item action-bar-item__sum" value="+">
-                    <input type="button" class="calculator-action-bar__item action-bar-item__result" value="=">
+                    <input type="button" class="calculator-action-bar__item action-bar-item__ce" id="ce" value="CE">
+                    <input type="button" class="calculator-action-bar__item action-bar-item__seven action-bar__numbers" value="7">
```
Comment (alexeymagdich) [inline index.html]: I would suggest using singular form for such class names (`.action-bar__number`, `.action-bar__operator` etc.)

File: js/script.js [hunk 19]
```diff
@@ -1,29 +1,65 @@
...
-    document.getElementById('result').innerHTML = 'result ' + c1;   
- } 
- 
- function calcSub() {
-    const a = +document.getElementById('a').value; 
-    const b = +document.getElementById('b').value;
-    const c2 = a - b;
-    document.getElementById('result').innerHTML = 'result ' + c2;   
- }
+let globalNumbers = document.querySelectorAll('.action-bar__numbers'),
```
Comment (alexeymagdich) [inline js/script.js]: Use `let` only for mutable variables. In all other cases you should use `const`. Read [this article](https://medium.com/javascript-scene/javascript-es6-var-let-or-const-ba58b8dcde75).

File: js/script.js [hunk 20]
```diff
@@ -1,29 +1,65 @@
...
+    operators = document.querySelectorAll('.action-bar__operators'),
```
Comment (alexeymagdich) [inline js/script.js]: Use separate declarations for each variable: 

```
const globalNumbers = ...;
const operators = ...;
let operatorClicked = ...;
etc.
```

It will improve code readability and maintenance.

File: js/script.js [hunk 21]
```diff
@@ -1,29 +1,65 @@
...
+    operatorClicked = false;
+    output.value = '0';
 
- function calcMul() {
-    const a = +document.getElementById('a').value; 
-    const b = +document.getElementById('b').value;
-    const c3 = a * b;
-    document.getElementById('result').innerHTML = 'result ' + c3;   
- }
+for (let i = 0; i < globalNumbers.length; i++) {
```
Comment (alexeymagdich) [inline js/script.js]: You can extract the initialization of handlers to a separate function:

```
function initialize() {
  const globalNumbers = document.querySelectorAll('.action-bar__numbers');
  for (let i = 0; i < globalNumbers.length; i++) { ... }

  const operators = document.querySelectorAll('.action-bar__operators');
  for (let i = 0; i < operators.length; i++) { ... }

  ...
}
```

Then you will have to call this function somewhere in the script (e.g. on `document.onload` event).

It will help to encapsulate initialization logic (with all necessary variables) in a one simple function.

File: js/script.js [hunk 22]
```diff
@@ -1,29 +1,65 @@
...
- }
+for (let i = 0; i < operators.length; i++) {
+    operators[i].addEventListener('click', event =>
+    operation(event.target.defaultValue)
+    );
+}
 
- 
+cleanEntry.addEventListener('click', event =>
+clean(event.target.defaultValue)
```
Comment (alexeymagdich) [inline js/script.js]: Function `clean` does not have parameters. Why are you sending `event.target.defaultValue` here?

File: js/script.js [hunk 23]
```diff
@@ -1,29 +1,65 @@
...
+);
+
+function numberOnClick(num) { 
+    
+    if (operatorClicked === true) {
```
Comment (alexeymagdich) [inline js/script.js]: It can be simplified. The following expressions:

- `if (operatorClicked === true)`
- `if (operatorClicked)`

are the same.

File: js/script.js [hunk 24]
```diff
@@ -1,29 +1,65 @@
...
+    operation(event.target.defaultValue)
+    );
+}
 
- 
+cleanEntry.addEventListener('click', event =>
+clean(event.target.defaultValue)
+);
+
+function numberOnClick(num) {
```
Comment (alexeymagdich) [inline js/script.js]: It's a bad name. Better name for this function is `onNumberClick`.

**Review by alexeymagdich — CHANGES_REQUESTED** [review]
This PR has terrible code styles. 

1. Please, ensure you installed `ESLint` extension.
2. Setup `ESLint` extension to auto-fix / auto-format your code on save.

File: js/script.js [hunk 25]
```diff
@@ -63,3 +54,65 @@ function clean() {
...
+    for (let i = 0; i < operators.length; i++) {
+        operators[i].addEventListener('click', event =>
+        operation(event.target.defaultValue)
+        );
+    }
+
+    const decimal = document.getElementById('dec');
+    decimal.addEventListener('click', event => 
+    decimalEntry(event.target.defaultValue)
+    );
```
Comment (alexeymagdich) [inline js/script.js]: Once again: fix all indentation problems. In this example, you have absolutely flat structure. Here's the correct indentation:

```
decimal.addEventListener('click', event => 
    decimalEntry(event.target.defaultValue)
);
```

You can beautify your code here (just copy and paste the whole file): https://beautifier.io/. It will help to understand correct indentation.

File: js/script.js [hunk 26]
```diff
@@ -63,3 +54,65 @@ function clean() {
...
+
+    const sqrt = document.getElementById('sqrt');
+    sqrt.addEventListener('click', () =>
```
Comment (alexeymagdich) [inline js/script.js]: Let's simplify this in the following way:

```
sqrt.onclick(roorSquare);
```

You can simplify all other same handlers in such way.

File: js/script.js [hunk 27]
```diff
@@ -1,47 +1,38 @@
...
-for (let i = 0; i < operators.length; i++) {
-    operators[i].addEventListener('click', event =>
-    operation(event.target.defaultValue)
-    );
-}
-
-cleanEntry.addEventListener('click', event =>
-clean(event.target.defaultValue)
-);
+window.onload = initialize;
```
Comment (alexeymagdich) [inline js/script.js:1]: Add one empty line after it for logic separation purposes.

**Review by alexeymagdich — APPROVED** [review]
LGTM.
